src/util/base64.rs

```diff
@@ -0,0 +1,274 @@
+//! Handling of Base 64-encoded data.
+//!
+//! This module provides various methods for decoding and encoding data in
+//! Base 64. Because there are different dialects of Base 64 and applications
+//! have slight usage differences atop those, the module provides an number
+//! of structs that that describe flavors of Base 64 used within in a certain
+//! context. That is, you don’t have to remember how an application uses
+//! Base 64 exactly but just pick your application.
+//!
+//! Each flavor implements a number of methods for encoding and decoding.
+//! These differ slightly between the flavors based on what they are used
+//! for.
+use std::{error, fmt, io, str};
+use std::io::Read;
+use base64::engine::{DecodeEstimate, Engine};
+use base64::engine::general_purpose::{
+    GeneralPurpose, GeneralPurposeConfig, STANDARD
+};
+
+pub use base64::{DecodeError, DecodeSliceError};
+
+
+//------------ Type Aliases --------------------------------------------------
+
+pub type DecoderReader<T> = base64::read::DecoderReader<
+    'static, GeneralPurpose, T
+>;
+
+
+//------------ Xml -----------------------------------------------------------
+
+/// The flavour used by `base64Binary` defined for XML.
+///
+/// This uses the standard alphabet with padding and ASCII white-space allowed
+/// between alphabet characters.
+pub struct Xml;
+
+impl Xml {
+    const ENGINE: GeneralPurpose = STANDARD;
```

Comment:
Also see the other comment on xml/decode.rs. Previously that was using `STANDARD_NO_PAD`. Are we sure the `STANDARD` here is what we want?